app/jobs/allocation_job.rb

```diff
@@ -2,38 +2,89 @@ class AllocationJob < ApplicationJob
   queue_as :default
 
   def perform(allocation_batch_job)
-    school = allocation_batch_job.school
-    order_state = allocation_batch_job.order_state
-    current_allocation = school.std_device_allocation
+    set_instance_variables(allocation_batch_job)
+    set_default_new_raw_allocation_value
+    set_default_new_raw_cap_value
 
-    current_allocation_value = current_allocation&.raw_allocation || 0
-    devices_ordered = current_allocation&.raw_devices_ordered || 0
-    new_allocation_value = [current_allocation_value + allocation_batch_job.allocation_delta, devices_ordered].max
+    # A negative delta must check against devices_available_to_order
+    set_negative_allocation_and_cap_values if @allocation_delta.negative?
 
-    current_cap_value = current_allocation&.raw_cap || 0
-    new_cap_value = [current_cap_value + allocation_batch_job.allocation_delta, devices_ordered].max
+    ActiveRecord::Base.transaction do
+      update_current_allocation!
+      call_school_order_state_and_cap_update_service
+      record_batch_job_processed_and_notify
+    end
+  end
 
-    disable_user_notifications = !allocation_batch_job.send_notification
+private
 
-    allocation = SchoolDeviceAllocation.find_or_initialize_by(school: school, device_type: 'std_device')
+  def set_allocation_batch_job
```

Comment:
Doesn't look like this is being used anymore (given the `set_instance_variables` method below)?

Reply:
Correct. Removed. This is all too hurried for my liking.